File: design-patterns/java/facade/VersionControlSystem.java
public class VersionControlSystem {
    public void pullLatestChanges(String branch) {
        System.out.println("VCS: Pulling latest changes from '" + branch + "'...");
        simulateDelay();
        System.out.println("VCS: Pull complete.");
    }
    
    private void simulateDelay() {
        try {
            Thread.sleep(1000);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }
}

File: design-patterns/python/facade/version_control_system.py
import time

class VersionControlSystem:
    def pull_latest_changes(self, branch: str) -> None:
        print(f"VCS: Pulling latest changes from '{branch}'...")
        self._simulate_delay()
        print("VCS: Pull complete.")
    
    def _simulate_delay(self) -> None:
        time.sleep(1) 